### DISCUSSION TITLE
How can I set two factors for a cell in grid environment such as land use type, DEM?

### DISCUSSION THREAD

**Discussion by QingshengYang** [JoelForamitti/agentpy#10] [Q&A] [answered]
Hi, everyone, How can I set two factors for a cell in grid environment such as land use type, DEM?

**Comment by jofmi** [comment]
Hi! Do you mean that you want to set attributes to the cell itself?

**Reply by QingshengYang** [reply]
Hi, Joel. Thank you very much. Yes, I want to set two attributes to the cell. How can I do it? Best, QS

**Reply by jofmi** [reply]
This is a feature that I am still working on, I will share it in this thread when it's ready!
Meanwhile, a work-around would be to put an agent in each cell and set the attribute to that agent!

**Reply by QingshengYang** [reply]
Thank you very much. Yes, I think the cell has more than 2 attributes ans agent also has more than 2 attributes. By the way, can I set the grid environment to a renctangle such as 50*70, if Yes, how do I set it? Best,QS

**Reply by jofmi** [reply]
This you can do by calling `self.add_grid([50, 70])` from within the model.

**Reply by QingshengYang** [reply]
Hi Joel, Thank you very much and I hope seing your new features. Best, QS

**Comment by jofmi** [comment, marked as answer]
Hi @QingshengYang, since AgentPy v0.1.0, the Grid class has a method `Grid.add_field()` that can be used for cell attributes. Find more about it here: https://agentpy.readthedocs.io/en/latest/reference_grid.html#agentpy.Grid.add_field

**Comment by QingshengYang** [comment]
Dear Joel，
&nbsp;
&nbsp;Thank you very much for your kindly reminder.
Regards，
Qingsheng
------------------&nbsp;Original&nbsp;------------------
From: &nbsp;"Joël ***@***.***&gt;;
Date: &nbsp;Mon, May 31, 2021 05:54 PM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: &nbsp;Re: [JoelForamitti/agentpy] How can I set two factors for a cell in grid environment such as land use type, DEM? (#10)

&nbsp;



 
Hi @QingshengYang, since AgentPy v0.1.0, the Grid class has a method Grid.add_field() that can be used for cell attributes. Find more about it here: https://agentpy.readthedocs.io/en/latest/reference_grid.html#agentpy.Grid.add_field
